solr/core/src/test/org/apache/solr/handler/admin/api/ClusterPropsAPITest.java

```diff
@@ -0,0 +1,178 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.handler.admin.api;
+
+import static org.apache.solr.common.util.Utils.getObjectByPath;
+
+import java.net.URL;
+import java.util.List;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.StringEntity;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.cloud.SolrCloudTestCase;
+import org.apache.solr.common.util.Utils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ClusterPropsAPITest extends SolrCloudTestCase {
+
+  private URL baseUrl;
+  private String baseUrlV2ClusterProps;
+
+  private static final String testClusterProperty = "ext.test";
+  private static final String testClusterPropertyValue = "test value";
+  private static final String testClusterPropertyNestedKeyAndValue =
+      "  \"defaults\": {"
+          + "    \"collection\": {"
+          + "      \"numShards\": 4,"
+          + "      \"nrtReplicas\": 2,"
+          + "      \"tlogReplicas\": 2,"
+          + "      \"pullReplicas\": 2"
+          + "    }"
+          + "  }";
+  private static final String testClusterPropertyBulkAndNestedValues =
+      "{ \"properties\": {"
+          + testClusterPropertyNestedKeyAndValue
+          + ","
+          + "  \""
+          + testClusterProperty
+          + "\": "
+          + "\""
+          + testClusterPropertyValue
+          + "\""
+          + " } }";
+
+  @BeforeClass
+  public static void setupCluster() throws Exception {
+    configureCluster(1).addConfig("conf", configset("cloud-minimal")).configure();
+  }
+
+  @Before
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+
+    baseUrl = cluster.getJettySolrRunner(0).getBaseUrl();
+    baseUrlV2ClusterProps =
+        cluster.getJettySolrRunner(0).getBaseURLV2().toString() + "/cluster/properties";
+  }
+
+  @After
+  @Override
+  public void tearDown() throws Exception {
+    super.tearDown();
+  }
+
+  @Test
+  public void testClusterPropertyOpsAllGood() throws Exception {
```

Comment:
[+1] Great tests - afaict we have very little coverage around cluster-props so this is a huge improvement!

[-0] These tests seem like a good candidate for using the SolrJ classes generated by your API definitions (i.e. `o.a.s.client.solrj.request.ClusterPropertiesApi`).

Doing so feels a bit circular (i.e. any bugs in the JAX-RS annotations would get propagated to the generated code and might not be caught)...but it'd help the tests to scan a lot nicer IMO, and it'd get us coverage for the SolrJ classes to boot, so it sees like a worthwhile tradeoff IMO.  Wdyt?